src/ui/FieldValue/FieldValue.ts

```diff
@@ -403,14 +403,16 @@ export class FieldValue extends Component {
 
     if (atLeastOneFacetIsEnabled) {
       const selected = find(facets, (facet: Facet) => {
-        const facetValue = facet.values.get(value);
+        const facetValue = facet.values.get(originalFacetValue);
         return facetValue && facetValue.selected;
       });
 
+      const label = selected ? l('RemoveFilterOn', renderedFacetValue) : l('FilterOn', renderedFacetValue);
       new AccessibleButton()
-        .withLabel(selected ? l('RemoveFilterOn', value) : l('FilterOn', value))
+        .withLabel(label)
+        .withTitle(label)
```

Comment:
Does the `label` need to be added as both the label and title? What does it sound like on a screen-reader? From what I remember, this will output `Remove filter on value button remove filter on value`.